### PR TITLE
Add Dutch adjective suffixes to make forms

### LINKED ISSUE #323: Add Dutch adjective suffixes
PR: https://github.com/Yoast/javascript/pull/325

We should attach adjective suffixes to stems of Dutch words in order to create adjective stems.

### PR DESCRIPTION
**Pull request by agnieszkaszuba**
## Summary

Adds adjective suffixes to regular Dutch adjectives.

## Test instructions

This PR can be tested by following these steps:

* Add more word stems to the spec, together with all of the expected word forms (excluding the stem). Make sure all the expected forms are created.
  * To get stems, run some adjectives to the stemmer to get stems first. E.g., from the input word `korte` you should get the stem `kort`.
  * Use the stem you got in the adjective spec. You should then get forms such as `korte`, `korter`, `kortst` etc., plus also some nonsense forms. (In principle, obtaining nonsense forms is expected behavior. But if you obtain lots of nonsense forms that are ambiguous with other words, this might lead to too many false positives. So also keep an eye on the nonsense forms.)

### Note: 
Please change the package.json premium data specification to `feature/Dutch-morphology` after merging this PR.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [x] I have added unittests to verify the code works as intended

Fixes #323

### CODE REVIEW

File: packages/yoastseo/src/morphology/dutch/addAdjectiveSuffixes.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+ */
+const modifyStem = function( stemmedWord, modificationGroup ) {
+	const neededReplacement = modificationGroup.find( replacement => stemmedWord.search( new RegExp( replacement[ 0 ] ) ) !== -1 );
+	let foundModification = false;
+
+	if ( typeof neededReplacement !== "undefined" ) {
+		stemmedWord = stemmedWord.replace( new RegExp( neededReplacement[ 0 ] ), neededReplacement[ 1 ] );
+		foundModification = true;
+
+		return { stemmedWord, foundModification };
```
Comment (nataliashitova) [inline packages/yoastseo/src/morphology/dutch/addAdjectiveSuffixes.js]: You do not need a `return` here: you've reassigned `stemmedWord` and `foundModification`, so the `return` on line 91 will do the job.

File: packages/yoastseo/src/morphology/dutch/addAdjectiveSuffixes.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+ */
+const findAndApplyModifications = function( stemmedWord, typeSuffix, morphologyDataNL ) {
+	const ieelToIelModification = modifyStem( stemmedWord, morphologyDataNL.addSuffixes.stemModifications.ieelToIel );
+	if ( ieelToIelModification.foundModification === true ) {
+		stemmedWord = ieelToIelModification.stemmedWord;
+		return stemmedWord;
+	}
+	const doubleConsonantModification = modifyStem( stemmedWord, morphologyDataNL.addSuffixes.stemModifications.doublingConsonant );
+	if ( doubleConsonantModification.foundModification === true ) {
+		stemmedWord = doubleFinalLetter( stemmedWord );
```
Comment (nataliashitova) [inline packages/yoastseo/src/morphology/dutch/addAdjectiveSuffixes.js]: You can easily include this "modification" in the `modifyStem`. Just add an `if` that checks if the modification is `doublingConsonant` and perform your modification, while in all other cases perform the `replace` modification that you are currently performing.

File: packages/yoastseo/src/morphology/dutch/addAdjectiveSuffixes.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+ * Creates the second stem of words that have two possible stems (this includes stem ending in -ieel or -iël;
+ * stem with double or single vowel; ending in double or single consonant; ending in s/f or z/v). The inflected and comparative
+ * suffixes are then added to the modified stem.
+ *
+ * @param {string} stemmedWord The stem
+ * @param {string} typeSuffix The type of suffix that should be added to the inputted stem.
+ * @param {object} morphologyDataNL The Dutch morphology data file
+ * @returns {string} The modified stem, or the original stem if no modifications were made.
+ */
+const findAndApplyModifications = function( stemmedWord, typeSuffix, morphologyDataNL ) {
```
[the remark this answers is not in the document]
Comment (nataliashitova) [reply packages/yoastseo/src/morphology/dutch/addAdjectiveSuffixes.js]: If you use my advice on changing `doubleFinalLetter` into one of the ways you `modifyStem`, you can significantly simplify the code down to something like this:
```
const modificationsToFind = [ "ieelToIel", "doublingConsonant", "consonantVoicing", "vowelUndoublingOther", "vowelUndoublingComparative" ];

for ( const modification in modificationsToFind ) {
	const modificationData = modifyStem( stemmedWord, morphologyDataNL.addSuffixes.stemModifications[ modification ] );

	if ( modificationData.foundModification ) {
		return modificationData.stemmedWord;
	}
} 
```

**Comment by hansjovis** [comment]
Acceptance test: ✅

**Pull request merged**